Source/WebCore/svg/SVGFEComponentTransferElement.cpp

```diff
@@ -75,23 +76,74 @@ void SVGFEComponentTransferElement::svgAttributeChanged(const QualifiedName& att
 
 RefPtr<FilterEffect> SVGFEComponentTransferElement::createFilterEffect(const FilterEffectVector&, const GraphicsContext&) const
 {
-    ComponentTransferFunction red;
-    ComponentTransferFunction green;
-    ComponentTransferFunction blue;
-    ComponentTransferFunction alpha;
-
-    for (auto& child : childrenOfType<SVGElement>(*this)) {
-        if (is<SVGFEFuncRElement>(child))
-            red = downcast<SVGFEFuncRElement>(child).transferFunction();
-        else if (is<SVGFEFuncGElement>(child))
-            green = downcast<SVGFEFuncGElement>(child).transferFunction();
-        else if (is<SVGFEFuncBElement>(child))
-            blue = downcast<SVGFEFuncBElement>(child).transferFunction();
-        else if (is<SVGFEFuncAElement>(child))
-            alpha = downcast<SVGFEFuncAElement>(child).transferFunction();
+    using Channel = FEComponentTransfer::Channel;
+
+    EnumeratedArray<Channel, ComponentTransferFunction, Channel::Alpha> functions;
+
+    for (auto& child : childrenOfType<SVGComponentTransferFunctionElement>(*this))
+        functions[child.channel()] = child.transferFunction();
+
+    return FEComponentTransfer::create(functions[Channel::Red], functions[Channel::Green], functions[Channel::Blue], functions[Channel::Alpha]);
+}
+
+static bool isRelevantTransferFunctionElement(const Element& child)
```

Comment:
I am not sure what this function returns. Does it take care of the case where multiple functions of the same type are children of  SVGComponentTransferFunctionElement, for example:

           <feComponentTransfer>
                 <feFuncR type="linear" slope="1" intercept="0"/>
                 <feFuncG type="linear" slope="1" intercept="0"/>
                 <feFuncR type="linear" slope="1" intercept="1"/>
             </feComponentTransfer>

Reply:
Yes, exactly. The last instance of a given feFuncX child should win.